### PR TITLE
Integrate detekt junit

### PR DESCRIPTION
**Pull request by BraisGabin**
After see #4699 I implemented the Rule [`MissingTestAnnotation`](https://github.com/BraisGabin/detekt-junit-rules/blob/main/src/main/kotlin/com/braisgabin/detekt/junit/MissingTestAnnotation.kt) to avoid this kind of issues.

The problem that I see si that this rule set is stored on Jitpack (I want to publish it on `mavenCental` but each time I think about all those steps...). What do you think? Is it ok to have a dependency from Jitpack?

### CODE REVIEW

**Comment by 3flex** [comment]
I really think Jitpack should be a last resort, and this should be published to Maven Central.

You will probably get more users if published there too.

**Comment by cortinico** [comment]
> So I'm 👍 to merging but I'd really like to see it published in a more appropriate repo first.

Same. If you move it under the github.com/detekt org, we could reuse some of the Github Actions and Keys setup to publish this as well

File: detekt-gradle-plugin/build.gradle.kts [hunk 1]
```diff
@@ -16,6 +16,7 @@ detekt {
 repositories {
     mavenCentral()
     google()
+    maven { url = uri("https://jitpack.io") }
```
Comment (Goooler) [inline detekt-gradle-plugin/build.gradle.kts:19]:
```suggestion
    maven("https://jitpack.io")
```

File: settings.gradle.kts [hunk 2]
```diff
@@ -69,5 +69,6 @@ gradleEnterprise {
 dependencyResolutionManagement {
     repositories {
         mavenCentral()
+        maven { url = uri("https://jitpack.io") }
```
Comment (Goooler) [inline settings.gradle.kts:72]:
```suggestion
        maven("https://jitpack.io")
```

**Comment by 3flex** [comment]
I _think_ that was just a suggestion to make publishing easier if you didn't want to go through the steps of setting it up yourself.

I agree that treating the rule set as a third party rule set is a good idea, and not to host in the detekt org, for reasons you outlined.

**Comment by BraisGabin** [comment]
I know, I know, there was 0 anger or bad vibes on my comment (sorry if it seemed so). I *speak* Kotlin better than English 😂. It was just my point of push the third-party plugins. But I appreciate the proposal of cortinico, it means that he likes the rules :)

File: settings.gradle.kts (hunk 2 above)
Comment (yogurtearl) [reply settings.gradle.kts:72]: a note for the future, should use `exclusiveContent` where ever possible. 
https://docs.gradle.org/current/userguide/declaring_repositories.html#declaring_content_exclusively_found_in_one_repository

**Comment by BraisGabin** [comment]
Closing until I upload it to maven central

**Pull request closed without merging**